spring-data-relational/src/main/java/org/springframework/data/relational/core/mapping/DefaultNamingStrategy.java

```diff
@@ -25,6 +25,14 @@
  */
 public class DefaultNamingStrategy implements NamingStrategy {
 
+
+	/**
+	 * Static instance of the class.
+	 * <p>
+	 * Using this avoids creating essentially the same class over and over again.
+	 */
+	public static NamingStrategy INSTANCE = new DefaultNamingStrategy();
```

Comment:
`DefaultNamingStrategy` is mutable, we should override the setter to make it immutable. Ideally, throw `UnsupportedOperationException` with a hint in the documentation.